nginx/ngx_js_fetch.c

```diff
@@ -948,24 +953,27 @@ ngx_js_method_process(njs_vm_t *vm, ngx_js_request_t *request)
         njs_null_str,
     };
 
+    str.start = request->method.data;
+    str.length = request->method.len;
+
     for (m = &forbidden[0]; m->length != 0; m++) {
-        if (njs_strstr_case_eq(&request->method, m)) {
```

Comment:
If we go this route with eliminating njs_str_t we should avoid here `njs_strstr_case_eq()` and `njs_str_t `.
Need to be fixed here and below.

Reply:
We could introduce in ngx_js.h something like

```
#define
ngx_js_strstr_case_eq(s1, s2)                                                    \      
    (((s1)->len == (s2)->len)                                           \      
     && (ngx_strncasecmp((s1)->start, (s2)->start, (s1)->length) == 0))  
```

Reply:
Hi Dmitry,
I feel the string-related APIs have room for improvement, and the same for mem pool.
Ieally, there is something like `ngx_js_str_t`, `ngx_js_mp_t`, and they are related APIs around them.
It's just a draft idea.